### PR TITLE
Upgrade MediaWiki to 1.31

### PR DESCRIPTION
**Pull request by magol**

### CODE REVIEW

**Comment by magol** [comment]
@legoktm ups, I have fixed that now

File: legacy/Dockerfile [hunk 1]
```diff
@@ -5,7 +5,7 @@ RUN apt-get update && apt-get install -y \
 		git \
 		imagemagick \
 		libicu-dev \
-		# Required for SytaxHighlighting
+		# Required for SyntaxHighlighting
```
Comment (davidbarratt) [inline legacy/Dockerfile]: Would you mind fixing this typo in the other Dockerfiles? :)

Comment (davidbarratt) [reply legacy/Dockerfile]: Nevermind, it looks correct in the other ones. :)

**Comment by magol** [comment]
@davidbarratt If no-one disagree, I agree that we should updating apcu to the latest version.

**Comment by davidbarratt** [comment]
Docker Hub PR
https://github.com/docker-library/official-images/pull/4463

**Comment by tianon** [comment]
Just for future reference, see https://github.com/docker-library/docs/tree/master/php#pecl-extensions for why pinning PECL to a specific version is recommended:

> It is *strongly* recommended that users use an explicit version number in their `pecl install` invocations to ensure proper PHP version compatibility (PECL does not check the PHP version compatiblity when choosing a version of the extension to install, but does when trying to install it).
> 
> For example, `memcached-2.2.0` has no PHP version constraints (https://pecl.php.net/package/memcached/2.2.0), but `memcached-3.0.4` requires PHP 7.0.0 or newer (https://pecl.php.net/package/memcached/3.0.4). When doing `pecl install memcached` (no specific version) on PHP 5.6, PECL will try to install the latest release and fail.
> 
> Beyond the compatibility issue, it's also a good practice to ensure you know when your dependencies receive updates and can control those updates directly.

**Pull request merged**